handler/src/main/java/io/netty/handler/timeout/IdleStateEvent.java

```diff
@@ -21,19 +21,14 @@
  * A user event triggered by {@link IdleStateHandler} when a {@link Channel} is idle.
  */
 public final class IdleStateEvent {
-    public static final IdleStateEvent FIRST_READER_IDLE_STATE_EVENT = new IdleStateEvent(IdleState.READER_IDLE, true);
```

Comment:
@jchambers - The general approach with 4.1 (before final is released ... which should be soon) is that you can add to the API but not change/remove existing APIs.  So this change would be a concern for 4.1 (and of course 4.0).  Not to say there aren't alternative approaches ... for example maybe extend this class (make it non-final) and translate the static idle events into non-static events (similar to what you have done here).